corgidrp/l2b_to_l3.py

```diff
@@ -19,20 +19,22 @@ def create_wcs(input_dataset, astrom_calibration):
     updated_dataset = input_dataset.copy()
 
     northangle = astrom_calibration.northangle
-    platescale = astrom_calibration.platescale
-    center_coord = astrom_calibration.boresight
+    platescale_x, platescale_y = astrom_calibration.platescale
+    ra_offset, dec_offset = astrom_calibration.avg_offset
 
     # create wcs for each image in the dataset
     for image in updated_dataset:
 
         im_data = image.data
-        image_shape = im_data.shape
-        center_pixel = [image_shape[1] // 2, image_shape[0] // 2]
+        image_y, image_x = im_data.shape
+        center_pixel = [(image_x-1) // 2, (image_y-1) // 2]
+        target_ra, target_dec = image.pri_hdr['RA'], image.pri_hdr['DEC']
+        corrected_ra, corrected_dec = target_ra - ra_offset, target_dec - dec_offset
         roll_ang = image.pri_hdr['ROLL']
 
         vert_ang = np.radians(northangle + roll_ang)  ## might be -roll_ang
         pc = np.array([[-np.cos(vert_ang), np.sin(vert_ang)], [np.sin(vert_ang), np.cos(vert_ang)]])
-        cdmatrix = pc * (platescale * 0.001) / 3600.
+        cdmatrix = pc * (platescale_x * 0.001) / 3600.  # assumes platescale is same along both axes
```

Comment:
I think pc[0,0] and pc[1,0] should be platescale_x and pc[0,1] and pc[1,1] should be platescale_y to not assume the same